attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -186,25 +235,72 @@ impl KbsClient<Box<dyn EvidenceProvider>> {
         Ok(())
     }
 
+    /// Convert the runtime data and the nonce into a hashed representation using the
+    /// specified hash algorithm.
+    async fn hash_runtime_data(
+        &self,
+        runtime_data: String,
+        nonce: String,
+        tee: Tee,
+        algorithm: HashAlgorithm,
+    ) -> Result<Vec<u8>> {
+        debug!("Hashing {tee:?} runtime data using nonce {nonce} and algorithm {algorithm:?}");
+
+        let hashed_data = match algorithm {
+            HashAlgorithm::Sha256 => {
+                let mut hasher = Sha256::new();
+
+                hasher.update(runtime_data);
+
+                match tee {
+                    // IBM SE uses nonce as runtime_data to pass attestation_request
+                    Tee::Se => nonce.into_bytes(),
+                    _ => hasher.finalize().to_vec(),
+                }
+            }
+            HashAlgorithm::Sha384 => {
+                let mut hasher = Sha384::new();
+
+                hasher.update(runtime_data);
+
+                match tee {
+                    // IBM SE uses nonce as runtime_data to pass attestation_request
+                    Tee::Se => nonce.into_bytes(),
+                    _ => hasher.finalize().to_vec(),
+                }
+            }
+            HashAlgorithm::Sha512 => {
+                let mut hasher = Sha512::new();
+
+                hasher.update(runtime_data);
+
+                match tee {
+                    // IBM SE uses nonce as runtime_data to pass attestation_request
+                    Tee::Se => nonce.into_bytes(),
+                    _ => hasher.finalize().to_vec(),
+                }
+            }
```

Comment:
We have `HashAlgorithm::digest()` function that could help to avoid the match arms here.